src/hotspot/share/gc/z/zHeapIterator.cpp

```diff
@@ -23,41 +23,79 @@
 
 #include "precompiled.hpp"
 #include "classfile/classLoaderData.hpp"
-#include "classfile/classLoaderDataGraph.hpp"
+#include "gc/shared/taskqueue.inline.hpp"
 #include "gc/z/zAddress.inline.hpp"
-#include "gc/z/zBarrier.inline.hpp"
 #include "gc/z/zGlobals.hpp"
 #include "gc/z/zGranuleMap.inline.hpp"
 #include "gc/z/zHeapIterator.hpp"
+#include "gc/z/zLock.inline.hpp"
 #include "gc/z/zOop.inline.hpp"
-#include "gc/z/zRootsIterator.hpp"
-#include "gc/z/zStat.hpp"
 #include "memory/iterator.inline.hpp"
 #include "utilities/bitMap.inline.hpp"
-#include "utilities/stack.inline.hpp"
 
 class ZHeapIteratorBitMap : public CHeapObj<mtGC> {
 private:
-  CHeapBitMap _map;
+  CHeapBitMap _bitmap;
 
 public:
   ZHeapIteratorBitMap(size_t size_in_bits) :
-      _map(size_in_bits) {}
+      _bitmap(size_in_bits, mtGC) {}
 
   bool try_set_bit(size_t index) {
-    if (_map.at(index)) {
-      return false;
+    return _bitmap.par_set_bit(index);
+  }
+};
+
+class ZHeapIteratorContext {
+private:
+  ZHeapIterator* const           _iter;
+  ZHeapIteratorQueue* const      _queue;
+  ZHeapIteratorArrayQueue* const _array_queue;
+  const uint                     _worker_id;
+  ZStatTimerDisable              _timer_disable;
```

Comment:
Why `ZStatTimerDisable`? Which timer(s) is it suppressing? I see that there's another one in `ZHeapIterator` as well.

Reply:
It's suppressing uses of `ZStatTimer` in the root iterators.

Reply:
Got it; thank you.